### PR TITLE
Made Sampler and InferenceParameters threadsafe

### PR DESCRIPTION
**Pull request by LLukas22**

### CODE REVIEW

**Review by danforbes — APPROVED** [review]
Looks very simple and was informally approved by @philpax in Discord, so I feel safe merging.

File: crates/llm-base/src/lib.rs [hunk 1]
```diff
@@ -85,6 +85,11 @@ pub struct InferenceParameters {
     /// sampler that offers a [Default](samplers::TopPTopK::default) implementation.
     pub sampler: Arc<dyn Sampler>,
 }
+
+//Since Sampler implements Send and Sync, InferenceParameters should too.
+unsafe impl Send for InferenceParameters {}
```
Comment (philpax) [inline crates/llm-base/src/lib.rs:90]: hol up is this necessary? won't these be inherited?

Comment (philpax) [reply crates/llm-base/src/lib.rs:90]: (it's fine but I don't think it's needed?)

Comment (LLukas22) [reply crates/llm-base/src/lib.rs:90]: PyO3 was complaining, so i had to wrap it in an Arc<>. With this i can just use it in a python objekt.

Comment (philpax) [reply crates/llm-base/src/lib.rs:90]: weird, I think it should automatically infer Send + Sync for inference parameters as all the fields should be, too 🤔 

It's totally fine as is, but can you try playing around with it and see if it still compiles without the unsafe impls?

**Pull request merged**